### ISSUE TITLE
problem of train cnn model

### ISSUE THREAD

**Issue by banbsyip** [sassoftware/python-dlpy#240]
I set up a  cnn model to classify mnist data.the model as follow:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/18277076/67650863-98b70100-f979-11e9-85c9-c9b4007ece29.png">
i train the model firstly successful,the result as follow:
![image](https://user-images.githubusercontent.com/18277076/67650958-e3d11400-f979-11e9-9775-45ff1a09ea02.png)
![image](https://user-images.githubusercontent.com/18277076/67650964-e92e5e80-f979-11e9-9bff-c897bea959be.png)
when i changed the characters ,if the max_epochs greater than  step_size, the train go wrong.
<img width="706" alt="image" src="https://user-images.githubusercontent.com/18277076/67651346-80e07c80-f97b-11e9-94dd-d4fc1870a57e.png">
<img width="702" alt="image" src="https://user-images.githubusercontent.com/18277076/67651026-2f83bd80-f97a-11e9-85b0-3f3f5b8ff898.png">
<img width="686" alt="image" src="https://user-images.githubusercontent.com/18277076/67651037-3f9b9d00-f97a-11e9-9ec7-c48db52f8460.png">
<img width="638" alt="image" src="https://user-images.githubusercontent.com/18277076/67651051-4de9b900-f97a-11e9-956f-6489f8008993.png">
<img width="582" alt="image" src="https://user-images.githubusercontent.com/18277076/67651064-56da8a80-f97a-11e9-8d59-2524d1570f11.png">

**Comment by ethem-kinginthenorth** [comment]
learning rate schedulers are created for easy support to learning rate modification.  we will have checks to prevent a smaller step_size than the max_epochs when lr_schedulers are used.

**Comment by banbsyip** [comment]
> learning rate schedulers are created for easy support to learning rate modification. we will have checks to prevent a smaller step_size than the max_epochs when lr_schedulers are used.

when I trained the tinyyolo, the same problem occurred.the code from https://go.documentation.sas.com/?cdcId=pgmsascdc&cdcVersion=9.4_3.4&docsetId=casdlpg&docsetTarget=p1np8zbnoyd0brn1dhehthuuxj4q.htm&locale=zh-CN#p1dgtuox2j70uen1sg8qpg203n9z
it is about yolo's object detection
![image](https://user-images.githubusercontent.com/18277076/67760905-292a3a00-fa7d-11e9-924e-7a34d608736f.png)
![image](https://user-images.githubusercontent.com/18277076/67760947-43fcae80-fa7d-11e9-811e-1e9a4e84ffd7.png)
![image](https://user-images.githubusercontent.com/18277076/67760977-4f4fda00-fa7d-11e9-8941-5cacf0e08130.png)
![image](https://user-images.githubusercontent.com/18277076/67761000-5aa30580-fa7d-11e9-83bb-ed51618530b0.png)

**Comment by ethem-kinginthenorth** [comment]
thanks for pointing out. I will be looking into this today.

**Comment by ethem-kinginthenorth** [comment]
@banbsyip seems like there might be an issue with the connection or such. would you please set "swat.options.cas.debug_responses = True " and then re-run your code. I would like to see what the server sends back

**Comment by banbsyip** [comment]
> eems like there might be an issue with the connection or such. would you please set "swat.options.cas.debug.responses = True" and then re-run your code. I would like to see what the server sends back

ok,i will run the code as your suggestion. I have a question about the label,i use the voc2007 data,the label of "person" have some parts such as "head","left","hand"and so on. I label all the parts as "person", is it a correct way to label?

**Comment by sasxianhu** [comment]
Depends on what your task is. If you want to detect the parts, then label them for sure. If you think any human parts should use person as label, then label them as person.

**Comment by banbsyip** [comment]
> Depends on what your task is. If you want to detect the parts, then label them for sure. If you think any human parts should use person as label, then label them as person.

thanks for your help.

**Comment by banbsyip** [comment]
> @banbsyip seems like there might be an issue with the connection or such. would you please set "swat.options.cas.debug.responses = True" and then re-run your code. I would like to see what the server sends back
the result as follow:
![image](https://user-images.githubusercontent.com/18277076/67823042-874a3200-fafc-11e9-894b-55bed4f46dc2.png)

**Comment by ethem-kinginthenorth** [comment]
@banbsyip thanks for sharing the log message. I am communicating with swat folks to see if they can help resolve this issue.

**Comment by Guma02** [comment]
Hello, I know this is an old forum, but I am having the same error message. Does anyone know the solution? @banbsyip @ethem-kinginthenorth Thanks in advance.
